Lib/asyncio/timeouts.py

```diff
@@ -0,0 +1,141 @@
+import enum
+
+from types import TracebackType
+from typing import final, Any, Dict, Optional, Type
+
+from . import events
+from . import tasks
+
+
+__all__ = (
+    "Timeout",
+    "timeout",
+    "timeout_at",
+)
+
+
+class _State(enum.Enum):
+    CREATED = "created"
+    ENTERED = "active"
+    EXPIRING = "expiring"
+    EXPIRED = "expired"
+    EXITED = "finished"
+
+
+@final
+class Timeout:
+
+    def __init__(self, when: Optional[float]) -> None:
+        self._state = _State.CREATED
+
+        self._timeout_handler: Optional[events.TimerHandle] = None
+        self._task: Optional[tasks.Task[Any]] = None
```

Comment:
```suggestion
        self._task: Optional[tasks.Task] = None
```
I got a complaint from pyright (VS Code + pylance + pyright in strict mode) because Task is generic in typeshed but not in the stdlib, and this being part of the stdlib it resolves to the stdlib Task. We can make it slightly more happy by leaving out the `[Any]` -- that's the default anyways.